src/lib/formats/rpk.h

```diff
@@ -0,0 +1,149 @@
+// license:LGPL-2.1+
+// copyright-holders:Michael Zapf
+/***************************************************************************
+
+	rpk.h
+
+	RPK format support
+
+***************************************************************************/
+
+#ifndef MAME_FORMATS_RPK_H
+#define MAME_FORMATS_RPK_H
+
+#pragma once
+
+#include "unzip.h"
+#include "hash.h"
+#include "xmlfile.h"
+
+#include <exception>
+#include <optional>
+
+namespace util {
+/***************************************************************************
+	TYPE DEFINITIONS
+***************************************************************************/
+
+class archive_file;
```

Comment:
You shouldn’t be forward-declaring stuff from libutils here – `#include "utilfwd.h"` if you want the forward declarations.